api/envoy/config/route/v3/route_components.proto

```diff
@@ -750,10 +750,49 @@ message RouteAction {
     ConnectConfig connect_config = 3;
   }
 
+  message ClusterSelector {
```

Comment:
Within the selector, the types should be a required oneof, right?

Also, re: my comment below about fallback routing, I would suggest that we plan for this selector to eventually have other state provided such as when a cluster is valid. For example, it's current health state, etc. wdyt?

Reply:
Yes, it should be a `oneof`. I've just pushed a new commit addressing this.

> Also, re: my comment below about fallback routing, I would suggest that we plan for this selector to eventually have other state provided such as when a cluster is valid. For example, it's current health state, etc. wdyt?

Yeah, this SGTM. I'll put more thought into this proto's structure with that in mind.